### PR TITLE
Migrate to play 2.5

### PR DESCRIPTION
**Pull request by mario-galic**
Migrated from Play 2.4 to 2.5 by following [this guide](https://www.playframework.com/documentation/2.5.x/Migration25). Key changes are

* removal of global objects
* everything is injected including `WSClient`, `MessageApi`, etc.

Migrated [play-googleauth](https://github.com/guardian/play-googleauth/) from 0.3 .to 0.6 which is designed for Play 2.5. Classes such as `GoogleAuthConf`, `GoogleGroups`, etc. have been removed as their functionality is now provided out of the box by 0.6. I followed the [example for 0.6](https://github.com/guardian/play-googleauth/tree/b7efbea6d9d4f366166858da9b090123cfe3bf72/example). 

Removed custom logic for session expiry as it is adding no extra security since if the attacker has physical access to machine and the session is expired all they have to do is just click on the particular Google Account - they are not asked to re-enter the password. It seems that `max_auth_age` in play-googleauth is not actually working: https://groups.google.com/forum/#!topic/google-federated-login-api/lKxJf4cZEcY

The secrets file now specifies all the necessary groups the user needs to be in as a list:

```
groups {
  useradmin = ["example@example.co.uk","2fa_enforce@example.co.uk"]
}
```

Hence, you will need to get a fresh DEV secretes file:
```
aws s3 cp --profile identity s3://gu-identity-admin-private/DEV/identity-admin.conf /etc/gu
```

### CODE REVIEW

File: app/auth/AuthActions.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  // Google configuration
+  override val authConfig = GoogleAuthConfig(
+    clientId     = conf.clientId,
+    clientSecret = conf.clientSecret,
+    redirectUrl  = conf.redirectUrl,
+    domain       = "guardian.co.uk"
+  )
+
+  // your app's routing
```
Comment (jacobwinch) [inline app/auth/AuthActions.scala]: Can the comments from the example be removed?

File: app/config/Conf.scala [hunk 2]
```diff
@@ -1,42 +1,81 @@
...
+  //  Google Auth
+
+  val clientId = configuration.getString("identity-admin.google.clientId")
+    .getOrElse(throw new IllegalStateException("Missing configuration: identity-admin.google.clientId"))
+
+  val clientSecret = configuration.getString("identity-admin.google.clientSecret")
+    .getOrElse(throw new IllegalStateException("Missing configuration: identity-admin.google.clientSecret"))
+
+  val redirectUrl = configuration.getString("identity-admin.google.authorisationCallback")
+    .getOrElse(throw new IllegalStateException("Missing configuration: dentity-admin.google.authorisationCallback"))
```
Comment (jacobwinch) [inline app/config/Conf.scala]: Very minor point - should be 'identity'

File: app/services/AdminApiHealthCheck.scala [hunk 3]
```diff
@@ -37,4 +44,12 @@ class AdminApiHealthCheck(adminApi: AdminApi, actorSystem: ActorSystem) extends
       triggerUpdate()
     }
   }
+
+  if (conf.stage == "PROD") {
+    logger.info("Starting Admin API HealthCheck agent...")
+    start()
+  }
+  else {
```
Comment (jacobwinch) [inline app/services/AdminApiHealthCheck.scala]: What's the advantage of avoiding running the healthcheck in DEV?

File: build.sbt [hunk 4]
```diff
@@ -7,15 +7,15 @@ version := "1.0.0-SNAPSHOT"
...
   "org.mockito" % "mockito-core" % "1.9.5" % "test",
-  "com.gu" %% "play-googleauth" % "0.3.3",
-  "org.webjars" %% "webjars-play" % "2.4.0-1",
+  "com.gu" %% "play-googleauth" % "0.6.0",
+  "org.webjars" %% "webjars-play" % "2.4.0-2",
   "org.webjars" % "bootstrap" % "3.3.7-1",
   "org.webjars" % "font-awesome" % "4.7.0",
-  "com.typesafe.akka" %% "akka-agent" % "2.4.0",
-  "com.typesafe.akka" %% "akka-slf4j" % "2.4.0",
+  "com.typesafe.akka" %% "akka-agent" % "2.5.4",
```
Comment (jacobwinch) [inline build.sbt]: I guess in a future PR we should look at replacing agents, perhaps with: https://github.com/guardian/box

File: conf/logback.xml [hunk 5]
```diff
@@ -36,7 +36,7 @@
         <appender-ref ref="ACCESSLOGFILE"/>
     </logger>
 
-    <root level="WARN">
+    <root level="INFO">
```
Comment (jacobwinch) [inline conf/logback.xml]: 👍

File: app/services/AdminApiHealthCheck.scala (hunk 3 above)
Comment (mario-galic) [reply app/services/AdminApiHealthCheck.scala]: In this PR I tried just to migrate Play while keeping pre-existing business logic as much as possible. So this AdminApiHealthCheck is how it was before.

File: build.sbt (hunk 4 above)
Comment (mario-galic) [reply build.sbt]: Yeap, definitely.

**Pull request merged**